src/genie/libs/parser/iosxe/show_interface.py

```diff
@@ -3220,42 +3220,60 @@ def cli(self, output=None):
         # Gi1/2     TelenlqPOIU        notconnect   125          full    100 10/100/1000-TX
-        # Gi1/1/0/1      FAST-HELLO         connected    4094       a-full a-1000 10/100/1000BaseTX
+        # Gi1/3     SE                 connected    132        a-full a-1000 10/100/1000-TX
+        # Gi1/7                        notconnect   99           auto   auto 10/100/1000-TX
+        # Gi1/10    To cft123          connected    trunk      a-full a-1000 10/100/1000-TX
+        # Gi1/1/0/1 FAST-HELLO         connected    4094       a-full a-1000 10/100/1000BaseTX
         # Te1/1/2   VSL                connected    trunk        full  a-10G 10GBase-SR
-        # Po1       ES1SW80AUN6        connected    trunk      a-full  a-10G
+        # Te2/1/20                     disabled     1            full   auto No XCVR
+        # Te2/1/21  VSL LINK1          disabled     1            full   auto No XCVR
+        # Po10      VSL LINK2          connected    trunk      a-full  a-10G
 
-        if out:
-            res = parsergen.oper_fill_tabular(device_output=out,
-                                              device_os='iosxe',
-                                              table_terminal_pattern=r"^\n",
-                                              header_fields=
-                                               [ "Port",
-                                                 "Name",
-                                                 "Status",
-                                                 "Vlan",
-                                                 "Duplex",
-                                                 "Speed",
-                                                 "Type"],
-                                              label_fields=
-                                              ["Interface",
-                                               "name",
-                                               "status",
-                                               "vlan",
-                                               "duplex_code",
-                                               "port_speed",
-                                               "type"],
-                                              index=[0]
-                                              )
-
-            #Building the schema out of the parsergen output
-            if res.entries:
-                 for intf, intf_dict in res.entries.items():
-                     intf = Common.convert_intf_name(intf)
-                     del intf_dict['Interface']
-                     parsed_dict.setdefault('interfaces', {}).update({intf: intf_dict})
 
-        return (parsed_dict)
+        # Name and Type are optional and also both can have multiple words on it, so find a single regex to match
+        # all cases is hard. I remove the Type column of the line and then the regex works for all cases
+        p1 = re.compile(r'^(?P<interfaces>[\d\w/]+)(?: +(?P<name>([\S ]+)))?'
```

Comment:
Hi @salvapinyol ,

Optional keys can be handled by regex by using parentheses plus a question mark, like `(this_is_an_optional_key)?` 

```
Python 3.6.8 (default, Aug 20 2019, 17:12:48)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import re

>>> r = re.compile(r'^(?P<interfaces>\S+)(?: +(?P<name>([\S ]+)))? +(?P<status>(connected|notconnect|inactive|disabled|err-disabled|monitoring)) +(?P<vlan>\S+) +(?P<duplex_code>[\S\-]+) +(?P<port_speed>[\S\-]+)( +(?P<type>.+))?$')

>>> no_type = 'Po10      VSL LINK2          connected    trunk      a-full  a-10G'

>>> has_type = 'Gi1/1/0/1 FAST-HELLO         connected    4094       a-full a-1000 10/100/1000BaseTX'

>>> no_type_dicts = r.match(no_type).groupdict()

>>> has_type_dicts = r.match(has_type).groupdict()

>>> no_type_dicts
{'interfaces': 'Po10', 'name': 'VSL LINK2         ', 'status': 'connected', 'vlan': 'trunk', 'duplex_code': 'a-full', 'port_speed': 'a-10G', 'type': None}

>>> has_type_dicts
{'interfaces': 'Gi1/1/0/1', 'name': 'FAST-HELLO        ', 'status': 'connected', 'vlan': '4094', 'duplex_code': 'a-full', 'port_speed': 'a-1000', 'type': '10/100/1000BaseTX'}
>>>
```

Overall, your code looks good to go! The comments in line#3237 and line#3238 can be removed now ;D
Thanks!